internal/common/actions/create_source_dir/create_source_dir.js

```diff
@@ -0,0 +1,181 @@
+const fs = require("fs-extra");
+const path = require("path");
+const { BazelAction, ensureArray } = require("../run_js/BazelAction");
+
+/**
+ * Action for creating a directory with the passed in files for symlinking and
+ * copying.
+ *
+ * The files can follow a uri syntax with params:
+ * recurse=false (default)
+ * symlink=true (default)
+ *
+ *
+ * The files to be populated have flags set on them for the appropriate actions:
+ * [flags]:./file/path
+ * Possible flags:
+ * s: Symlink
+ *   s:./some/dir/
+ * c: Copy
+ *   c:/some/file
+ * r: Recurse
+ *   rs:./some/dir/
+ * m: Module (put it in node_modules)
+ * mr: Recursive module (add all the modules to node_modules)
+ * g: Generate (run the passed in script to generate source files
+ *
+ * eg. node create_source_dir.js s:file/to/symlink.
+ */
+BazelAction(
+  {
+    string: [
+      // Root path to symlink/copy from
+      "from",
+      // The folder to populate
+      "into"
+    ]
+  },
+  async args => {
+    const { current_target, workspace_name, package_path, from, into } = args;
+    const sources = ensureArray(args._);
+    // console.log(sources);
```

Comment:
Remove this? (same below)

Alternatively, should we have a global `DEBUG` constant we can switch on/off?